examples/ode.py

```diff
@@ -0,0 +1,112 @@
+"""
+Predator-Prey Model
+===================
+
+This example replicates the great case study [1], which leverages the Lotka-Volterra
+equation [2] to describe the dynamics of Canada lynx (predator) and snowshoe hare
+(prey) populations. We will use the dataset obtained from [3] and run MCMC to get
+inferences about parameters of the differential equation governing the dynamics.
+
+**References:**
+
+    1. Bob Carpenter (2018), `"Predator-Prey Population Dynamics: the Lotka-Volterra model in Stan"
+       <https://mc-stan.org/users/documentation/case-studies/lotka-volterra-predator-prey.html/>`_.
+    2. https://en.wikipedia.org/wiki/Lotka–Volterra_equations
+    3. http://people.whitman.edu/~hundledr/courses/M250F03/M250.html
+"""
+
+import argparse
+import os
+
+import matplotlib
+import matplotlib.pyplot as plt
+import pandas as pd
+
+from jax.experimental.ode import build_odeint
+import jax.numpy as np
+from jax.random import PRNGKey
+
+import numpyro
+import numpyro.distributions as dist
+from numpyro.infer import MCMC, NUTS, Predictive
+
+matplotlib.use('Agg')  # noqa: E402
+
+
+def dz_dt(z, t, alpha, beta, gamma, delta):
+    """
+    Lotka–Volterra equations. Real positive parameters `alpha`, `beta`, `gamma`, `delta`
+    describes the interaction of two species.
+    """
+    u = z[0]
+    v = z[1]
+    du_dt = (alpha - beta * v) * u
+    dv_dt = (-gamma + delta * u) * v
+    return np.stack([du_dt, dv_dt])
+
+
+predator_prey_int = build_odeint(dz_dt, rtol=1e-5, atol=1e-3, mxstep=500)
+
+
+def model(N, y=None):
+    """
+    :param int N: number of measurement times
+    :param numpy.ndarray y: measured populations with shape (N, 2)
+    """
+    # initial population
+    z_init = numpyro.sample("z_init", dist.LogNormal(np.log(10), 1), sample_shape=(2,))
+    # measurement times
+    ts = np.arange(float(N))
+    # parameters alpha, beta, gamma, delta of dz_dt
+    theta = numpyro.sample(
+        "theta",
+        dist.TruncatedNormal(low=0., loc=np.array([0.5, 0.05, 1.5, 0.05]),
+                             scale=np.array([0.5, 0.05, 0.5, 0.05])))
+    # integrate dz/dt, the result will have shape N x 2
+    z = predator_prey_int(z_init, ts, *theta)
+    # measurement errors, we expect that measured hare has larger error than measured lynx
+    sigma = numpyro.sample("sigma", dist.Exponential(np.array([1, 2])))
+    # measured populations (in log scale)
+    numpyro.sample("y", dist.Normal(np.log(z), sigma), obs=y)
+
+
+def main(args):
+    URL = "http://people.whitman.edu/~hundledr/courses/M250F03/LynxHare.txt"
+    data = pd.read_csv(URL, sep=" (?!$)", names=["year", "hare", "lynx"], index_col=0, engine="python")
```

Comment:
@neerajprad could you help me upload the following csv file (or .txt file) to the cloud?
```
import pandas as pd
URL = "http://people.whitman.edu/~hundledr/courses/M250F03/LynxHare.txt"
data = pd.read_csv(URL, sep=" (?!$)", names=["year", "hare", "lynx"], index_col=0, engine="python")
data.to_csv("LynxHare.csv")
```
Thanks! I'll write a reader for the file to remove pandas requirement in this example (currently, this makes test fail).